From e1b57d025b0523479fe906f050c5639ad280b481 Mon Sep 17 00:00:00 2001
From: Michael Stahl <mstahl@redhat.com>
Date: Thu, 7 Aug 2014 11:24:18 +0200
Subject: [PATCH] rhbz#1079672: FileDialogHelper: don't hand out stale preview
 Graphic

The maGraphic is updated from a timer, and if you double-click on a file
then the dialog may exit with the previous file still previewed in
maGraphic, so the wrong image is returned by the dialog.

Change-Id: I99094d85d8d68d5c8a842f52e7039dbbbf095995
(cherry picked from commit f1589f768bbd1d8baea6e442f392831851bdbb3b)
Reviewed-on: https://gerrit.libreoffice.org/10804
Reviewed-by: David Tardon <dtardon@redhat.com>
Tested-by: David Tardon <dtardon@redhat.com>
---
 sfx2/source/dialog/filedlghelper.cxx | 23 ++++++++++-------------
 1 file changed, 10 insertions(+), 13 deletions(-)

diff --git a/sfx2/source/dialog/filedlghelper.cxx b/sfx2/source/dialog/filedlghelper.cxx
index c8178a4..28a8060 100644
--- a/sfx2/source/dialog/filedlghelper.cxx
+++ b/sfx2/source/dialog/filedlghelper.cxx
@@ -789,23 +789,20 @@ ErrCode FileDialogHelper_Impl::getGraphic( Graphic& rGraphic ) const
 {
     ErrCode nRet = ERRCODE_NONE;
 
-    if ( ! maGraphic )
-    {
-        OUString aPath;;
-        Sequence < OUString > aPathSeq = mxFileDlg->getFiles();
+    // rhbz#1079672 do not return maGraphic, it need not be the selected file
 
-        if ( aPathSeq.getLength() == 1 )
-        {
-            aPath = aPathSeq[0];
-        }
+    OUString aPath;;
+    Sequence<OUString> aPathSeq = mxFileDlg->getFiles();
 
-        if ( !aPath.isEmpty() )
-            nRet = getGraphic( aPath, rGraphic );
-        else
-            nRet = ERRCODE_IO_GENERAL;
+    if (aPathSeq.getLength() == 1)
+    {
+        aPath = aPathSeq[0];
     }
+
+    if (!aPath.isEmpty())
+        nRet = getGraphic(aPath, rGraphic);
     else
-        rGraphic = maGraphic;
+        nRet = ERRCODE_IO_GENERAL;
 
     return nRet;
 }
-- 
1.9.3

